### PR TITLE
Raise Exception if `bbox_params` is not specified and bbox transformation is called

### PR DESCRIPTION
**Pull request by VirajBagal**
In response to this issue: https://github.com/albumentations-team/albumentations/issues/861

Made changes in `_check_args` method in `albumentations/core/composition.py`.

### CODE REVIEW

File: albumentations/core/composition.py [hunk 1]
```diff
@@ -253,6 +254,8 @@ def _check_args(self, **kwargs):
                 if data:
                     if not isinstance(data[0], np.ndarray):
                         raise TypeError("{} must be list of numpy arrays".format(data_name))
+            if internal_data_name in check_bbox_param and self.processors.get("bboxes") is None:
+                raise Exception("bbox_params must be specified for bbox transformations")
```
Comment (Dipet) [inline albumentations/core/composition.py:258]: Better to change `Exception` to `ValueError`.

**Comment by Dipet** [comment]
Please, also add this test into `tests/test_core`

```python
def test_bbox_params_is_not_set(image, bboxes):
    t = Compose([])
    with pytest.raises(ValueError) as exc_info:
        t(image=image, bboxes=bboxes)
    assert str(exc_info.value) == "bbox_params must be specified for bbox transformations"
```

File: albumentations/core/composition.py (hunk 1 above)
Comment (VirajBagal) [reply albumentations/core/composition.py]: `Exception` changed to `ValueError`

**Comment by VirajBagal** [comment]
> Please, also add this test into `tests/test_core`
> 
> ```python
> def test_bbox_params_is_not_set(image, bboxes):
>     t = Compose([])
>     with pytest.raises(ValueError) as exc_info:
>         t(image=image, bboxes=bboxes)
>     assert str(exc_info.value) == "bbox_params must be specified for bbox transformations"
> ```


Added the mentioned test.

**Comment by Dipet** [comment]
Tests found problems when we combine `Compose`, `OneOf`, or another such transform.
For example look to `test_transform_pipeline_serialization_with_bboxes`.

We need to add some logic to disable `_check_args` for all transforms inherited from `BaseCompose`

**Comment by VirajBagal** [comment]
`test_transform_pipeline_serialization_with_bboxes` has this example of augmentation: 
`    aug = A.Compose(
        [
            A.OneOrOther(
                A.Compose([A.RandomRotate90(), A.OneOf([A.HorizontalFlip(p=0.5), A.VerticalFlip(p=0.5)])]),
                A.Compose([A.Rotate(p=0.5), A.OneOf([A.HueSaturationValue(p=0.5), A.RGBShift(p=0.7)], p=1)]),
            ),
            A.SomeOf(
                [
                    A.HorizontalFlip(p=1),
                    A.Transpose(p=1),
                    A.HueSaturationValue(p=0.5),
                    A.RandomBrightnessContrast(p=0.5),
                ],
                n=5,
            ),
        ],
        bbox_params={"format": bbox_format, "label_fields": ["labels"]},
    )
`

This augmentation raises error because `A.Compose` within `A.OneOrOther` doesn't have the `bbox_params` defined. 

We have two options now:
1.  Either we make the user enter `bbox_params` wherever he/she uses `A.Compose`. We would then need to make these changes in every test where `A.OneOrOther` and bboxes are used. 
2.  Or we come up with a logic to use `bbox_params` defined in the outermost `A.Compose`  and use the same in the inner `A.Compose` of `A.OneOrOther`. 

What do you think? 

I tried the test with this and it works: 

`aug = A.Compose(
    [
        A.OneOrOther(
            A.Compose([A.RandomRotate90(), A.OneOf([A.HorizontalFlip(p=0.5), A.VerticalFlip(p=0.5)])], bbox_params={"format": bbox_format, "label_fields": ["labels"]}),
            A.Compose([A.Rotate(p=0.5), A.OneOf([A.HueSaturationValue(p=0.5), A.RGBShift(p=0.7)], p=1)], bbox_params={"format": bbox_format, "label_fields": ["labels"]}),
        ),
        A.SomeOf(
            [
                A.HorizontalFlip(p=1),
                A.Transpose(p=1),
                A.HueSaturationValue(p=0.5),
                A.RandomBrightnessContrast(p=0.5),
            ],
            n=5,
        ),
    ],
    bbox_params={"format": bbox_format, "label_fields": ["labels"]},
)
`

**Comment by Dipet** [comment]
I think is better to skip `_check_args` in inner transforms.
To do this we must check all transforms and if the transform is inherited from `Compose` to disable this check.
Something like this:
```python
class Compose:
    def __init__(...):
        ....
        self.is_check_args = True
        for transform in self.transforms.transforms:
            if issubclass(transform, Compose):
                transform._disable_check_args()
    
    def _disable_check_args(self):
            self.is_check_args = False

    def __call__(self, *args, force_apply=False, **data):
        if self.is_check_args:
            self._check_args(**data)
```

**Comment by VirajBagal** [comment]
@Dipet thanks for the guidance. Made the changes. Please take a look.

File: albumentations/core/composition.py [hunk 2]
```diff
@@ -170,10 +171,25 @@ def __init__(self, transforms, bbox_params=None, keypoint_params=None, additiona
 
         self.add_targets(additional_targets)
 
+        self.is_check_args = True
```
Comment (Dipet) [inline albumentations/core/composition.py:174]: Please add this var into `_to_dict`.
And reformat code with `black` https://albumentations.ai/docs/contributing/

File: albumentations/core/composition.py [hunk 3]
```diff
@@ -210,12 +263,15 @@ def _check_data_post_transform(self, data):
         return data
 
     def _to_dict(self):
+        self.is_check_args = True
```
Comment (Dipet) [inline albumentations/core/composition.py:266]: Oh my fault. We don't need this var there, because it is not an argument in `__init__`. Sorry.

**Comment by VirajBagal** [comment]
@Dipet why aren't the checks getting completed?

**Comment by Dipet** [comment]
Please pull the recent master branch and resolve conflicts

**Comment by VirajBagal** [comment]
Recreating new PR.

**Pull request closed without merging**